vim_menutrans_merger.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import os
+import re
+
+# Translate menu name to another language. See ":help :menutranslate".
+MENUTRANS_COMMANDS = {"tmenu", "menut", "menutrans", "menutranslate"}
+
+def make_translated_dict(translation_file, translated_dict):
+    """Get the translated files' content and put it into translated_dict"""
+
+    # Encode with "latin1", because we don't care about the correctness of 
+    # non-ASCII character.
+    with open(translation_file, encoding='UTF-8') as f:
+        for line_number, line in enumerate(f):
+            line = line.strip()
+
+            if not line:
+                continue
+
+            # Split a line by white spaces but not backslash-escaped spaces.
+            new_word_list = re.findall(r"(?:\\ |[^ \t])+", line)
+            match_index = [each for each in re.finditer(r"(?:\\ |[^ \t])+", line)]
+            
+            if len(new_word_list) > 1 and new_word_list[0] in MENUTRANS_COMMANDS:
+                translated_dict[new_word_list[1].lower()] = (
+                        line_number, translation_file, new_word_list[1], line[match_index[1].end():])
+            elif len(new_word_list) > 1 and new_word_list[0].startswith("g:menutrans_"):
+                translated_dict[new_word_list[0].lower()] = (
+                        line_number, translation_file, new_word_list[0], line[match_index[0].end():])
+            elif (len(new_word_list) > 1 and new_word_list[0] == "let" and 
+                    new_word_list[1].startswith("g:menutrans_")):
+                translated_dict[new_word_list[1].lower()] = (
+                        line_number, translation_file, new_word_list[1], line[match_index[2].end():])
+
+def extract_translated_message(template_file, translation_file):
+    """Get the translated files' content and put it into translated_dict"""
+
+    translated_dict = {}
+    make_translated_dict(translation_file, translated_dict)
+    replace_template_translation(template_file, translated_dict)
+
+def replace_template_translation(template_file, translated_dict):
+    """Replace the translations in template_file with the content in translated_dict"""
+
+    newContent = ""
+    # Encode with "latin1", because we don't care about the correctness of 
+    # non-ASCII character.
+    with open(template_file, encoding='UTF-8') as f:
```

Comment:
`UTF-8`? You're going to fail a lot of non-UTF-8 files.

Reply:
But we only care about UTF-8 files, because other different coded format files are generated from UTF-8 files.